apps/ledger-live-mobile/src/screens/Newsfeed/index.tsx

```diff
@@ -0,0 +1,169 @@
+import { InformativeCard, Flex, Text } from "@ledgerhq/native-ui";
+import React, { memo, useCallback, useMemo } from "react";
+import { FlatList, Linking, TouchableOpacity } from "react-native";
+import { TouchableHighlight } from "react-native-gesture-handler";
+import styled, { useTheme } from "styled-components/native";
+import { InAppBrowser } from "react-native-inappbrowser-reborn";
+import { useTranslation } from "react-i18next";
+import { track, TrackScreen } from "../../analytics";
+import { CryptopanicNewsWithMetadata } from "../../hooks/newsfeed/cryptopanicApi";
+import { inAppBrowserDefaultParams } from "../../components/InAppBrowser";
+import { useCryptopanicPosts } from "../../hooks/newsfeed/useCryptopanicPosts";
+import CryptopanicIcon from "../../icons/Cryptopanic";
+import Button from "../../components/wrappedUi/Button";
+import Skeleton from "../../components/Skeleton";
+import { ScreenName } from "../../const";
+import FormatRelativeTime from "../../components/FormatRelativeTime";
+
+const keyExtractor = (item: CryptopanicNewsWithMetadata) => item.id.toString();
+
+const imageNewsProps = {
+  style: { width: 90, height: 75 },
+};
+
+const CRYPTOPANIC_URL = "https://cryptopanic.com";
+const CRYPTOPANIC_DEFAULT_PARAMS = {
+  metadata: true,
+  approved: true,
+  public: true,
+};
+
+function NewsfeedPage() {
```

Comment:
You would certainly benefit in terms of performance (because I saw in the code of the hook that you had perf issues) by splitting this component in several components that consume the state from one/several context (or possibly even better from the redux store).

For instance a header, footer, empty state and the main list, and each item would be its own memoized component. And if you do that properly the main component would have barely any state in it, each part would be updating just when needed.